cpp/oneapi/dal/algo/decision_forest/detail/infer_ops.cpp

```diff
@@ -15,20 +15,45 @@
 *******************************************************************************/
 
 #include "oneapi/dal/algo/decision_forest/detail/infer_ops.hpp"
+#include "oneapi/dal/algo/decision_forest/parameters/cpu/infer_parameters.hpp"
 #include "oneapi/dal/algo/decision_forest/backend/cpu/infer_kernel.hpp"
 #include "oneapi/dal/backend/dispatcher.hpp"
 
 namespace oneapi::dal::decision_forest::detail {
 namespace v1 {
 
-template <typename Policy, typename Float, typename Task, typename Method>
-struct infer_ops_dispatcher<Policy, Float, Task, Method> {
-    infer_result<Task> operator()(const Policy& policy,
-                                  const detail::descriptor_base<Task>& desc,
+template <typename Context, typename Float, typename Task, typename Method>
+struct infer_ops_dispatcher<Context, Float, Task, Method> {
+    infer_result<Task> operator()(const Context& context,
+                                  const descriptor_base<Task>& desc,
+                                  const infer_parameters<Task>& params,
                                   const infer_input<Task>& input) const {
+        return implementation(context, desc, params, input);
+    }
+
+    infer_parameters<Task> select_parameters(const Context& ctx,
+                                             const descriptor_base<Task>& desc,
+                                             const infer_input<Task>& input) const {
+        using kernel_dispatcher_t = dal::backend::kernel_dispatcher< //
+            KERNEL_SINGLE_NODE_CPU(parameters::infer_parameters_cpu<Float, Method, Task>)>;
+        return kernel_dispatcher_t{}(ctx, desc, input);
+    }
+
+    infer_result<Task> operator()(const Context& context,
+                                  const descriptor_base<Task>& desc,
+                                  const infer_input<Task>& input) const {
+        const auto params = select_parameters(context, desc, input);
+        return implementation(context, desc, params, input);
+    }
+
+private:
+    inline auto implementation(const Context& context,
```

Comment:
Should naming be aligned with other hp algos? for example in cov/linreg it names const Policy& ctx, its little bit confusing